src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala

```diff
@@ -24,163 +29,187 @@ case class CromwellRuntimeAttributes(docker: Option[String],
     value match {
-      case Some(v) =>
-        val nels = v.split(",\\s*") map validateLocalDisk
-        val emptyDiskNel = Vector.empty[Disk].successNel[String]
-        val disksNel = nels.foldLeft(emptyDiskNel)((acc, v) => (acc |@| v) { (a, v) => a :+ v })
-        disksNel map {
-          case disks if disks.exists(_.getName == LocalDiskName) => disks
-          case disks => disks :+ LocalizationDisk
-        }
-      case None => Defaults.Disk.successNel
+      case Some(WdlBoolean(b)) => b.successNel
+      case Some(WdlString(s)) if s.toLowerCase == "true" => true.successNel
+      case Some(WdlString(s)) if s.toLowerCase == "false" => false.successNel
+      case Some(_) => s"Expecting ${FAIL_ON_STDERR.key} runtime attribute to be a Boolean or a String with values of 'true' or 'false'".failureNel
+      case None => defaults.failOnStderr.successNel
     }
   }
 
-  private def validateLocalDisk(disk: String): ValidationNel[String, Disk] = {
-    disk.split("\\s+") match {
-      case Array(name, DiskType.LOCAL.diskTypeName) =>
-        LocalDisk(name, DiskType.LOCAL).toDisk.successNel[String]
-      case Array(name, sizeGb, diskType) if diskType != DiskType.LOCAL.diskTypeName =>
-        (validateLong(sizeGb) |@| validateDiskType(diskType)) { (s, dt) => LocalDisk(name, dt, Option(s)).toDisk }
-      case _ => s"'$disk' should be in form 'NAME SIZE TYPE', with SIZE blank for LOCAL, otherwise SIZE in GB".failureNel
+  private def validateContinueOnReturnCode(value: Option[WdlValue]): ErrorOr[ContinueOnReturnCode] = {
+    value match {
+      case Some(b: WdlBoolean) => ContinueOnReturnCodeFlag(b.value).successNel
+      case Some(WdlString(s)) if s.toLowerCase == "true" => ContinueOnReturnCodeFlag(true).successNel
+      case Some(WdlString(s)) if s.toLowerCase == "false" => ContinueOnReturnCodeFlag(false).successNel
+      case Some(WdlInteger(i)) => ContinueOnReturnCodeSet(Set(i)).successNel
+      case Some(WdlArray(wdlType, seq)) =>
+        val nels = seq map validateInt
+        val defaultReturnCodeNel = Set.empty[Int].successNel[String]
+        nels.foldLeft(defaultReturnCodeNel)((acc, v) => (acc |@| v) { (a, v) => a + v }) map ContinueOnReturnCodeSet
+      case Some(_) => s"Expecting ${CONTINUE_ON_RETURN_CODE.key} runtime attribute to be either a Boolean, a String 'true' or 'false', or an Array[Int]".failureNel
+      case None => defaults.continueOnReturnCode.successNel
     }
   }
 
-  private def validateDiskType(diskTypeName: String): ValidationNel[String, DiskType] = {
-    DiskType.values().find(_.diskTypeName == diskTypeName) match {
-      case Some(diskType) => diskType.successNel[String]
-      case None =>
-        val diskTypeNames = DiskType.values.map(_.diskTypeName).mkString(", ")
-        s"Disk TYPE $diskTypeName should be one of $diskTypeNames".failureNel
+  private def validateLocalDisks(value: Option[WdlValue]): ErrorOr[Seq[Disk]] = {
```

Comment:
Honestly I tried to do this, but my editor is completely unresponsive trying to modify this block of code because of all the `scalaz` stuff.  I've already spent 20 minutes on this trying to get it to work so I'm going to drop it unless everybody thinks it's really important...  my half-baked work in progress code is here: https://gist.github.com/scottfrazer/5f2bb0333eee002ce0a1.